Add support for rails and set of fiducials and tooling holes

yaqwsx opened the pull request:
I implemented the requested features from #41 and #46. 

@martiinezz and @quixoticgeek can I ask you to try it on your designs to check if it works as expected? The documentation is not updated yet, but the help command is. Basically you specify `--tooling <horizontalOffset> <verticalOffset> <diameter>` and `--fiducials <horizontalOffset> <verticalOffset> <copperDiameter> <openingDiameter>`.

If you are interested in testing, install development version of KiKit by:
```
pip3 install git+https://github.com/yaqwsx/KiKit@rail
```

martiinezz commented:
I'll check it out tonight :)

martiinezz commented:
I see that it creates top/bottom rails only? Is there a way to create side rails too?

yaqwsx commented:
Yes, there is only the option to create the top and bottom rails. I was thinking about having both, but IMHO it overcomplicated the CLI (which is already fairly complicated). If you need side rails, why not just rotate your designs 90 degrees (there is an option to that). I plan to document this recommended procedure.

yaqwsx commented:
If more people find the need for side rails, I add the support. But I am worried the CLI will be even scarier (since there will be two commands for rails - hrail and vrail) and the same for fiducals and tooling holes.

iharshadec commented:
Earlier my assembler never asked for side rails. But recently looks like he changed some of his machines and requested for rails and so I used to get this request to my PCB fabricator. 
You are right @yaqwsx CLI won't look good but I think having options may be OK.

yaqwsx commented:
Would be a two step solution (first, create a panel with top rails, then rotate the whole panel by a second command) acceptable for you?

martiinezz commented:
I would have to rotate first and then add rails but why not add extra argument for --rails like tb for top/bottom and lr for sides?
So you could do that in one step if I select option tb nothing changes when I select lr - rotate first and add rails and the best would be to rotate back.

yaqwsx commented:
Ok, I added to two options for side and top rails. Fortunately, you need a single command for fiducials no matter which rails orientation you use.

I also added rails to the examples page.

martiinezz commented:
Looks way better now but I get 3 holes in 4 corners - right bottom corner is left intact.

![image](https://user-images.githubusercontent.com/19732099/89716642-02efde00-d9af-11ea-9299-2e0d0e3efbf2.png)

with --fiducials 2 2 1 2 --tooling 2 2 1.5 --railsLr 4

yaqwsx commented:
This is intended behavior - all documents I found about fiducials and tooling holes required 2 or 3 holes/fiducial, so it is easy to see the intended board orientation. If you place 4 holes, then you can easily make a mistake and rotate the panel.

martiinezz commented:
I don't have manufacturer requiring 4 holes. I have mostly used manufacturer service to make those tooling stuff.
Anyway this version seems to work for me. Is one mousebite per board side also standard for rails?

yaqwsx commented:
Could you explain what do you mean by "mouse bite"?

martiinezz commented:
Connection between board and rail - I get just one per 45 mm board.

![image](https://user-images.githubusercontent.com/19732099/89718813-6f280d00-d9c2-11ea-95db-b30a76c3f329.png)

yaqwsx commented:
These pieces are usually called "tabs". You can specify any number of them via `--vtabs` and `--htabs`. Please, read the documentation for details.

martiinezz commented:
OK I didn't know this is taken into account for rails too :) - anyway new code seems to work great now. Thanks !!!

iharshadec commented:
@yaqwsx is right. Acutally those tooling hole either they use fro exact stencil matching or something to hold the panel and there should be 3 only. Not on all sides.
